Bottom: f8c4f71b82758fd3ae8f0061f59baf6d67afafce
Top:    35bc44ac292a76995db6619da691c747a3cc9c78
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-24 20:19:15 +0000

Calculate epsilon estimate for debugging


---

diff --git a/src/polygon.c b/src/polygon.c
index 1694548..1901d82 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -123,6 +123,7 @@ void
 polygon_init (void)
 {
   struct rlimit limit;
+  double eps;
 
   double cos_ang = cos (2.0 * M_PI / POLY_CIRC_SEGS_F);
   double sin_ang = sin (2.0 * M_PI / POLY_CIRC_SEGS_F);
@@ -137,6 +138,12 @@ polygon_init (void)
   getrlimit (RLIMIT_AS, &limit);
   limit.rlim_cur = MIN (limit.rlim_cur, 2000 * 1024 * 1024 /* 2 GiB limit to virtual memory size */);
   setrlimit (RLIMIT_AS, &limit);
+
+  /* Estimate eps on this machine */
+  eps = 1.;
+  do eps = 0.5 * eps; while (eps + 1.0 > 1.0);
+
+  fprintf (stderr, "Epsilon calculated at %.17e\n", eps);
 }
 
 Cardinal
